### DISCUSSION TITLE
Documentation or Inclusion of other algorithms

### DISCUSSION THREAD

**Discussion by BradKML** [DerwenAI/pytextrank#251] [Ideas]
The models and algorithms in https://github.com/boudinfl/pke#implemented-models are similar to Textrank but not sped up by SpaCy, so it might be a good idea to include them in PyTextRank

PS: There are also other non TextRank-esque algorithms to consider when making this assessment:
- RAKE https://github.com/aneesha/RAKE and https://github.com/csurfer/rake-nltk and https://github.com/vgrabovets/multi_rake and https://github.com/chinwuDebug/RAKE_improve
- YAKE https://github.com/LIAAD/yake
- Aho–Corasick algorithm https://github.com/dav009/flash
- RaKUn https://github.com/Parsely/serpextract

**Comment by louisguitton** [comment]
thanks for bringing our attention to `pke` ! 

this issue is similar to #78 for which we have made already great progress with 2 contributions:
- adding `PositionRank` and `BiasedRank` 
- adding `BaseTextRank` and `BaseTextRankFactory` to enable integration of more flavours

Regarding the graph based models of `pke`, I can see this:

- their `TextRank` can be achieved with our `BaseTextRank(edge_weight=0)`
- their `SingleRank` can be achieved with our `BaseTextRank()` or `BaseTextRank(edge_weight=1.0)`
- their `PositionRank` can be achieved with our `PositionRank`

the following ones are missing:

* TopicRank [paper by (Bougouin et al., 2013)](http://aclweb.org/anthology/I13-1062.pdf)
* TopicalPageRank [article by (Sterckx et al., 2015)](http://users.intec.ugent.be/cdvelder/papers/2015/sterckx2015wwwb.pdf)
* MultipartiteRank [article by (Boudin, 2018)](https://arxiv.org/abs/1803.08721)

I was not aware of these 3 papers and approaches so thank you.
Do you have experience with them in practice and are they good?
Would you be open to contribute them?

**Comment by BradKML** [comment]
I am mainly reporting them for notes in Documentation, but if I can I would contribute

Also some extra note: https://github.com/miso-belica/sumy/blob/master/docs/alternatives.md
- Bipartite HITS https://github.com/himanshujindal/Automatic-Text-Summarizer
- LexRank https://github.com/giorgosera/pythia/blob/dev/analysis/summarization/summarization.py https://github.com/kylehg/summarizer
- topic models https://github.com/bobflagg/Topic-Networks
- MEAD http://www.summarization.com/mead/
- Luhn Summ https://github.com/talha1503/Extractive_Text_Summarization/blob/master/luhn_sum.py
- SumBasic https://github.com/talha1503/Extractive_Text_Summarization/blob/master/SumBasic.ipynb

**Comment by BradKML** [comment]
To reiterate the current algorithms that are not included:
- [ ] SumBasic by [Nankova et. al.](https://www.cs.bgu.ac.il/~elhadad/nlp09/sumbasic.pdf) and its [Repository in Python](https://github.com/talha1503/Extractive_Text_Summarization)
- [ ] LexRank by [Erkan et. al.](https://arxiv.org/pdf/1109.2128.pdf) and its [Repository in Python](https://github.com/giorgosera/pythia/blob/dev/analysis/summarization/summarization.py)
- [ ] SalianceRank [teneva et. al.](https://aclanthology.org/P17-2084.pdf) by and its [Reposiroty in Python](https://github.com/zhengfeitian/saliencerank)
- [ ] KEA by [Witten et. al.](http://community.nzdl.org/kea/Nevill-et-al-1999-DL99-poster.pdf) and its [Repository in Java](https://github.com/EUMSSI/KEA)
- [ ] UniKeyPhrase by [Wu et. al.](https://arxiv.org/pdf/2106.04847.pdf) and its [Repository in Python](https://github.com/thinkwee/UniKeyphrase)
- [ ] https://github.com/boudinfl/pke#implemented-models
  - [ ] TopicRank paper by (Bougouin et al., 2013)
  - [ ] TopicalPageRank article by (Sterckx et al., 2015)
  - [ ] MultipartiteRank article by (Boudin, 2018)

Looking at
- [ ] JAKE https://github.com/xcjackpan/jake
- [ ] Crackr https://github.com/anjishnu/Crackr

**Comment by ceteri** [comment]
Also check the algorithms listed in `pke` <https://github.com/boudinfl/pke>  which has an excellent range of implementations. FWIW, that library is GPL and not implemented as a spaCy pipeline, so there's some room for algorithm implementations both there (for research) and here (for production deployments).
